src/ol/renderer/layerrenderer.js

```diff
@@ -95,6 +97,33 @@ ol.renderer.Layer.prototype.handleImageChange = function(event) {
 
 
 /**
+ * Load the image if not already loaded, and manage the
+ * registration/deregistration of the image change listener.
+ * @param {ol.ImageBase} image Image.
+ * @return {boolean} `true` if the image is already loaded, `false`
+ *     otherwise.
+ * @protected
+ */
+ol.renderer.Layer.prototype.loadImage = function(image) {
+  var imageState = image.getState();
+  if (imageState == ol.ImageState.LOADED ||
+      imageState == ol.ImageState.ERROR) {
+    goog.events.unlisten(image, goog.events.EventType.CHANGE,
+        this.handleImageChange_, false, this);
```

Comment:
For a given image and a given layer renderer we do want the listener to be called just once, so `listenOnce` is indeed what we want to use here.

Just like `listen`, `listenOnce` will not create a new listener when it is called again with the same (target/image, listener, scope/renderer, capture) tuple.

So I'll try to use `listenOnce` again.